packages/grid/x-data-grid/src/components/base/GridBody.tsx

```diff
@@ -147,25 +85,6 @@ function GridBody(props: GridBodyProps) {
 
   return (
     <GridMainContainer ref={rootRef}>
-      <rootProps.slots.columnHeaders
```

Comment:
With column headers moved to the virtual scroller, it will be possible to make individual header cells sticky during horizontal scroll 👍🏻

https://github.com/mui/mui-x/assets/13808724/24c43a03-a090-42a5-ba33-a8da9e5426f9

This might be especially helpful with multi-column pivoting to make the column group structure clearer (see demo without sticky headers https://codesandbox.io/s/trusting-shtern-6tg36d

```diff
diff --git a/docs/data/data-grid/column-groups/BasicGroupingDemo.tsx b/docs/data/data-grid/column-groups/BasicGroupingDemo.tsx
index ea48cee7d..35b58e13e 100644
--- a/docs/data/data-grid/column-groups/BasicGroupingDemo.tsx
+++ b/docs/data/data-grid/column-groups/BasicGroupingDemo.tsx
@@ -6,18 +6,18 @@ const columns: GridColDef[] = [
   {
     field: 'firstName',
     headerName: 'First name',
-    width: 150,
+    width: 550,
   },
   {
     field: 'lastName',
     headerName: 'Last name',
-    width: 150,
+    width: 550,
   },
   {
     field: 'age',
     headerName: 'Age',
     type: 'number',
-    width: 110,
+    width: 550,
   },
 ];
 
@@ -61,6 +61,15 @@ export default function BasicGroupingDemo() {
         checkboxSelection
         disableRowSelectionOnClick
         columnGroupingModel={columnGroupingModel}
+        sx={{
+          '& .MuiDataGrid-columnHeaderTitleContainer': {
+            overflow: 'visible',
+          },
+          '& .MuiDataGrid-columnHeaderTitleContainerContent': {
+            position: 'sticky',
+            left: 8,
+          },
+        }}
       />
     </div>
   );

```